runtime/gc_vlhgc/AllocationContextBalanced.cpp

```diff
@@ -355,30 +358,37 @@ MM_AllocationContextBalanced::lockedAllocateArrayletLeaf(MM_EnvironmentBase *env
 	/* look up the spine region since we need to add this region to its leaf list */
 	MM_HeapRegionDescriptorVLHGC *spineRegion = (MM_HeapRegionDescriptorVLHGC *)_heapRegionManager->tableDescriptorForAddress(spine);
 	/* the leaf requires a pointer back to the spine object so that it can verify its liveness elsewhere in the collector */
-	leafAllocateData->setSpine(spine);
-	freeRegionForArrayletLeaf->resetAge(env, (U_64)_subspace->getBytesRemainingBeforeTaxation());
-	/* add the leaf to the spine region's leaf list */
-	/* We own the lock on the spine region's context when this call is made so we can safely manipulate this list.
-	 * An exceptional scenario: A thread allocates a spine (and possibly a few arraylets), but does not complete the allocation. A global GC (or a series of regular PGCs) occurs
-	 * that age out regions to max age. The spine moves into a common context. Now, we successfully resume the leaf allocation, but the common lock that
-	 * we already hold is not sufficient any more. We need to additionally acquire common context' common lock, since multiple spines from different ACs could have come into this state,
-	 * and worse multiple spines originally allocated from different ACs may end up in a single common context region.
-	 */
+	if (!MM_GCExtensions::getExtensions(env)->isVirtualLargeObjectHeapEnabled) {
```

Comment:
this should be below just before we add/push region to the list
the only thing that we change is that list is AC local (nor array local)
same problem of spine changing context to common likely exists with offheap
```
	if (!MM_GCExtensions::getExtensions(env)->isVirtualLargeObjectHeapEnabled)
			leafAllocateData->addToArrayletLeafList(spineRegion);
        else 
        		leafAllocateData->pushRegionToLeafRegionList(env, &_leafRegionList);
```